src/lib.rs

```diff
@@ -64,10 +72,66 @@ pub fn generate_keypair() -> KeyPair {
     JsValue::from(ret).into()
 }
 
+/// Derives `x25519` public key from a private key.
+///
+/// # Inputs
+///
+/// * `secret_key`: `x25519` private key.
+///
+/// # Returns
+///
+/// * `public_key`: `x25519` public key.
+///
 #[wasm_bindgen]
 pub fn derive_public(secret_key: &str) -> Result<JsString, JsValue> {
     let mut secret = [0; KEY_SIZE];
     let _ = b64_decode(secret_key.as_bytes(), &mut secret).map_err(|_| JsError::new("Base64"))?;
-    let public = PublicKey::from(secret);
+    let secret = StaticSecret::from(secret);
+    let public = PublicKey::from(&secret);
+    Ok(encode_key(&public.to_bytes()))
+}
+
+/// Converts `ed25519` public key to `x25519` public key.
+///
+/// # Inputs
+///
+/// * `public_key`: `ed25519` public key.
+///
+/// # Returns
+///
+/// * `public_key`: `x25519` public key.
+///
+#[wasm_bindgen]
+pub fn decompress_public(public_key: &str) -> Result<JsString, JsValue> {
+    let mut public = [0; KEY_SIZE];
+    let _ = b64_decode(public_key.as_bytes(), &mut public).map_err(|_| JsError::new("Base64"))?;
+    let compressed = CompressedEdwardsY::from_slice(&public[..]);
+    let public = compressed
+        .decompress()
+        .ok_or_else(|| JsError::new("Base64"))?
+        .to_montgomery();
+    Ok(encode_key(&public.to_bytes()))
+}
+
+/// Converts `x25519` public key to `ed25519` public key.
+///
+/// # Inputs
+///
+/// * `public_key`: `x25519` public key.
+/// * `sign`: a `u8` donating the desired sign of the resulting
+///   `EdwardsPoint`.  `0` denotes positive and `1` negative.
```

Comment:
I'm not too familiar with this crypto operation. Is it normal to accept this parameter, or would a boolean/string enum/-1 or 1 fit better?

Reply:
The difference indeed is small, one byte-shift away on the last byte.
In JavaScrip the default will be `0` anyway.

Reply:
I see what you're asking. This should be of `byte` type.

Reply:
Some standards will require it to be `0`. In a call `compress_public(key)` it will also be `0` by default. It works as optional parameter.